app/vmselect/netstorage/netstorage.go

```diff
@@ -1981,14 +1983,17 @@ func startStorageNodesRequest(qt *querytracer.Tracer, sns []*storageNode, denyPa
 	return &storageNodesRequest{
 		denyPartialResponse: denyPartialResponse,
 		resultsCh:           resultsCh,
+		qtp:                 qt,
 		qts:                 qts,
 		sns:                 sns,
 	}
 }
 
 func (snr *storageNodesRequest) finishQueryTracers(msg string) {
-	for qt := range snr.qts {
-		snr.finishQueryTracer(qt, msg)
+	for qt, storageAddr := range snr.qts {
+		cancelledQt := snr.qtp.NewChild("rpc at vmstorage: %s: %s", storageAddr, msg)
```

Comment:
it would be great to have a comment explaining what's happening here

Reply:
Added comment.